### PR TITLE
drivers: crypto: add driver for nRF ECB

### PR DESCRIPTION
**Pull request by maciekfabia**
Add driver for nRF AES Electronic CodeBook (ECB) peripheral

Signed-off-by: Maciej Fabia <maciej.fabia@nordicsemi.no>

### CODE REVIEW

**Review by anangl — APPROVED** [review]

File: CODEOWNERS [hunk 1]
```diff
@@ -138,6 +138,7 @@
 /drivers/clock_control/*nrf*              @nordic-krch
 /drivers/counter/                         @nordic-krch
 /drivers/counter/counter_cmos.c           @andrewboie
+/drivers/crypto/*nrf_ecb*                 @maciekfabia
```
Comment (ioannisg) [inline CODEOWNERS]: please add @anangl, too, in the list of CODEOWNERS for this entry

**Review by d3zd3z — APPROVED** [review]
This seems ok, at least as much as I can tell without documentation on how the hardware works. I'm also presuming that code using this would implement other modes in software, since ECB isn't directly safe to use within a protocol.

**Comment by rugeGerritsen** [comment]
Will you update the nRF BLE LL HAL to use this new driver interface? 
See https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/ecb.c
Some of the BLE crypto APIs should be wrapping this new driver: https://github.com/zephyrproject-rtos/zephyr/blob/master/include/bluetooth/crypto.h#L52

**Comment by ioannisg** [comment]
> Will you update the nRF BLE LL HAL to use this new driver interface?
> See https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/ecb.c
> Some of the BLE crypto APIs should be wrapping this new driver: https://github.com/zephyrproject-rtos/zephyr/blob/master/include/bluetooth/crypto.h#L52

Does this need to be done in this PR? Otherwise, a good question.

**Comment by rugeGerritsen** [comment]
> Will you update the nRF BLE LL HAL to use this new driver interface?
> See https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/ecb.c
> Some of the BLE crypto APIs should be wrapping this new driver: https://github.com/zephyrproject-rtos/zephyr/blob/master/include/bluetooth/crypto.h#L52
> 
> Does this need to be done in this PR? Otherwise, a good question.

I'm not sure if this will need to be done in this PR. If we don't do it now, we will have two driver interfaces for the ECB, one for BLE and one 'normal' driver interface.

**Comment by ioannisg** [comment]
>> I'm not sure if this will need to be done in this PR. If we don't do it now, we will have two driver interfaces for the ECB, one for BLE and one 'normal' driver interface.

I think this is fine; BLE can be updated later to use the new driver, with the help of @cvinayak

**Review by ioannisg — APPROVED** [review]

File: drivers/crypto/Kconfig.nrf_ecb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# NRF AES ECB configuration options
+
+# Copyright (c) 2020 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+
+menuconfig CRYPTO_NRF_ECB
+	bool "nRF AES electronic codebook mode encryption"
```
Comment (carlescufi) [inline drivers/crypto/Kconfig.nrf_ecb]: This is probably incompatible with the BLE controller, so we probably need to add a `depends on !BT_CTLR` or similar. cc @cvinayak

**Review by carlescufi — CHANGES_REQUESTED** [review]
I think we ne

File: drivers/crypto/Kconfig.nrf_ecb (hunk 2 above)
Comment (maciekfabia) [reply drivers/crypto/Kconfig.nrf_ecb]: done

**Review by cvinayak — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**